app/code/community/Bolt/Boltpay/controllers/ApiController.php

```diff
@@ -162,6 +162,25 @@ public function hookAction()
             );
             $this->getResponse()->setHttpResponseCode(201);
 
+        } catch (Bolt_Boltpay_DuplicatedTransitionException $boltPayDuplicatedTransitionException) {
+            //In this case, the parent quote is currently being processed or has been processed,
+            //and if this duplicated transaction has the same reference with the processed transaction, then it should be ignored
+            //and if not, there should report an error
+            if($boltPayDuplicatedTransitionException->getProcessedBoltReference() == $reference){
+                Mage::helper('boltpay/bugsnag')->notifyException( new Exception($e->getMessage()) );
+                $this->getResponse()->setHttpResponseCode(200);
+            }
+            else{
```

Comment:
@danac-gs, @daisy1754, @dumega --
Since,
- we are using the parent quote ID as the order reference
- and Bolt server-side has a 1-to-1 constraint on transaction reference and order reference

Then this case seems to be impossible to reach.  What am I missing here?  It is definitely good to have this case handled in the plugin code, but, if this is reached, it implies a bigger problem.  We would only expect the case of the transaction references being the same in the standard "racing condition scenario".

Second, @dumega I don't think that anything is set to `on-hold` here, which is a Magento order state.  I think the hook is simply failing and no merchant side actions are being taken, including order creation or any order state change.

Possibly, it is better to give an error message that identifies the transaction and order which are already in the Magento system which render this current transaction as erroneous and redundant.

Reply:
@mccottry the `on-hold` in error message is trying to explain there is no need to send duplicated webhook too soon, not a status signal. and if you look into it, the message already include related reference info.

`"The parent quote ". $immutableQuote->getParentQuoteId() ." is currently being processed or has been processed by bolt transaction ".$parentQuoteBoltReference."."`

and

`$e->getMessage().' Then this redundant bolt transaction '.$reference.' is set to on-hold'

Reply:
@dumega 
Maybe change the message to:
`$e->getMessage().' Therefore, this redundant bolt transaction '.$reference.' will not be processed as an order in Magento'